Uupdating `LD_LIBRARY_PATH` at runtime does not work

Joxit opened the pull request:
Hi @missinglink, what you did with #58 and #59 is incredible ! Now everything is working fine on my laptop ! :smiling_face_with_three_hearts:  But I still need some changes :sweat_smile:.

----

I had an error when installing libxml2.
```
Making install in python
make[2]: Entering directory '/tmp/libxml2/python'
make  install-recursive
make[3]: Entering directory '/tmp/libxml2/python'
Making install in .
make[4]: Entering directory '/tmp/libxml2/python'
make[5]: Entering directory '/tmp/libxml2/python'
make[5]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/opt/spatial/share/doc/libxml2-python-2.9.10'
 /usr/bin/install -c -m 644 TODO '/opt/spatial/share/doc/libxml2-python-2.9.10'
 /bin/mkdir -p '/usr/lib/python2.7/dist-packages'
 /usr/bin/install -c -m 644 drv_libxml2.py libxml2.py '/usr/lib/python2.7/dist-packages'
/usr/bin/install: cannot create regular file '/usr/lib/python2.7/dist-packages/drv_libxml2.py': Permission denied
/usr/bin/install: cannot create regular file '/usr/lib/python2.7/dist-packages/libxml2.py': Permission denied
```
The default python directory needs root privileges, but we are lucky we don't need python binding, so I added the option `--without-python`.

----

Then I ran a test `npm test` and I still got `SIGSEGV` even with the `RUNTIME` environment and the piece of code in `InitExtension.js`. So I added in `bin/test` the environment updates (with the correct `LD_LIBRARY_PATH` everything is working). I got the same issue with `node server/http.js`, so I added a new `bin/start`.

----

Then I tried `node bin/spatial.js serve` and I noticed that everything worked fine... I asked myself why ?
I found that when we use `bin/spatial.js`, it loads `InitExtension.js` (and update environment variables) and then spawn a new process with `LD_LIBRARY_PATH` correctly set. My guess is the node process should be spawned with the correct `LD_LIBRARY_PATH` environment.

That also means that `node bin/spatial.js import`, `node bin/spatial.js merge` and `node bin/spatial.js pip` do not work as-it.
I have two solutions and I would like your feedback @missinglink :

1. Move the environment code from `InitExtension.js` to `spatial.js` and spawn new processes for import, merge and pip (this implies instantiating `ImportService` from another file)
2. Encapsulate `bin/spatial.js` in a bash script which updates environment variables ?

----

I also added the environment `PELIAS_SPATIAL_RUNTIME` for `.bashrc` (less generic than `RUNTIME`)

missinglink commented:
heya, awesome glad you got it all working 🎉 

I'm still not really happy with how the dynamic linking works, setting `LD_LIBRARY_PATH` in every shell seems clunky and I'd like. to find a way to avoid it.

I've learned to much about compiling and installing C/C++ programs now ;)
The solution may lie in [setting the rpath](https://amir.rachum.com/blog/2016/09/17/shared-libraries/#rpath-and-runpath), I think this would mean that we no longer have to set `LD_LIBRARY_PATH`.

Regarding your PR.

`--without-python` is :+1: good catch

I would be fine with adding some sort of 'runtime init' script which is responsible for setting the env vars, but my preference would be to either:
- have it in a place where it didn't require going through a `bin/*` script (for instance if we run a single test case directly)
- not require setting those env vars at all (magically!?)

question for you?...
are you able to execute the `spatialite` binary from your local runtime *without* setting `LD_LIBRARY_PATH`?
```
/opt/spatial/bin/spatialite
```

because I'd also like for that binary (and others in the `${RUNTIME}/bin` dir) to magically know where to load their dependencies (maybe via `rpath`).

Joxit commented:
Yes, with all this configuration and `LD_LIBRARY_PATH`, the project seems more complicated... But all you did is a big step forward ! (no need of docker to run the project and disable tests for a simple commit... :sweat_smile:)

Okay, I updated the PR, I added a script for the CLI.
We can put all the environment configuration in a `.env` and source it ?

Yes it works :scream: ! 
```
joxit@laptop:~$ echo $LD_LIBRARY_PATH

joxit@laptop:~$ /opt/spatial/bin/spatialite
SpatiaLite version ..: 5.0.0-beta1	Supported Extensions:
	- 'VirtualShape'	[direct Shapefile access]
	- 'VirtualDbf'		[direct DBF access]
	- 'VirtualText'		[direct CSV/TXT access]
	- 'VirtualNetwork'	[Dijkstra shortest path]
	- 'RTree'		[Spatial Index - R*Tree]
	- 'MbrCache'		[Spatial Index - MBR cache]
	- 'VirtualSpatialIndex'	[R*Tree metahandler]
	- 'VirtualElementary'	[ElemGeoms metahandler]
	- 'VirtualKNN'	[K-Nearest Neighbors metahandler]
	- 'VirtualXPath'	[XML Path Language - XPath]
	- 'VirtualFDO'		[FDO-OGR interoperability]
	- 'VirtualGPKG'	[OGC GeoPackage interoperability]
	- 'VirtualBBox'		[BoundingBox tables]
	- 'SpatiaLite'		[Spatial SQL - OGC]
PROJ version ........: Rel. 7.0.1, May 1st, 2020
GEOS version ........: 3.8.1-CAPI-1.13.3
RTTOPO version ......: 1.1.0
TARGET CPU ..........: x86_64-linux-gnu
the SPATIAL_REF_SYS table already contains some row(s)
SQLite version ......: 3.31.1
Enter ".help" for instructions
SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for instructions
joxit@laptop:~$ ldd /opt/spatial/bin/spatialite
	linux-vdso.so.1 (0x00007ffe67feb000)
	libspatialite.so.7 => /opt/spatial/lib/libspatialite.so.7 (0x00007f17545cf000)
	libxml2.so.2 => /opt/spatial/lib/libxml2.so.2 (0x00007f1754470000)
	librttopo.so.1 => /opt/spatial/lib/librttopo.so.1 (0x00007f17543fc000)
	libreadosm.so.1 => /opt/spatial/lib/libreadosm.so.1 (0x00007f17543f1000)
	libproj.so.19 => /opt/spatial/lib/libproj.so.19 (0x00007f17540ff000)
	libtiff.so.5 => /opt/spatial/lib/libtiff.so.5 (0x00007f1754087000)
	libz.so.1 => /opt/spatial/lib/libz.so.1 (0x00007f1753c54000)
	libexpat.so.1 => /opt/spatial/lib/libexpat.so.1 (0x00007f1753b69000)
	libsqlite3.so => /opt/spatial/lib/libsqlite3.so (0x00007f1753a15000)
	libgeos_c.so.1 => /opt/spatial/lib/libgeos_c.so.1 (0x00007f17539d1000)
	libgeos-3.8.1.so => /opt/spatial/lib/libgeos-3.8.1.so (0x00007f17537fa000)
```

Now the LDD on qgis which use the same library as spatialite but with different versions
```
joxit@laptop:~$ ldd /usr/bin/qgis.bin 
	libproj.so.13 => /usr/lib/x86_64-linux-gnu/libproj.so.13 (0x00007fbfa8478000)
	libgeos_c.so.1 => /usr/lib/x86_64-linux-gnu/libgeos_c.so.1 (0x00007fbfa8440000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fbfa703a000)
	libsqlite3.so.0 => /usr/lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007fbfa6f18000)
	libspatialite.so.7 => /usr/lib/x86_64-linux-gnu/libspatialite.so.7 (0x00007fbfa6795000)
	libzip.so.4 => /usr/lib/x86_64-linux-gnu/libzip.so.4 (0x00007fbfa677a000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fbfa4d6d000)
	libxml2.so.2 => /usr/lib/x86_64-linux-gnu/libxml2.so.2 (0x00007fbfa3f92000)
	libgeos-3.7.1.so => /usr/lib/x86_64-linux-gnu/libgeos-3.7.1.so (0x00007fbfa36a0000)
	libtiff.so.5 => /usr/lib/x86_64-linux-gnu/libtiff.so.5 (0x00007fbfa1c6c000)
	libzstd.so.1 => /usr/lib/x86_64-linux-gnu/libzstd.so.1 (0x00007fbfa1bca000)
```

Joxit commented:
Since we are OK with this changes, I merge this PR :rocket:
